app/src/main/java/ai/elimu/analytics/task/UploadEventsWorker.kt

```diff
@@ -39,43 +40,56 @@ class UploadEventsWorker(context: Context, workerParams: WorkerParameters) :
+        for (learningEventDir in filesDir.listFiles() ?: emptyArray()) {
+            if (learningEventDir.name.startsWith("version-code-")) continue
+            if (learningEventDir.name != eventType.type) continue
 
-                        val baseApplication = applicationContext as BaseApplication
-                        val retrofit = baseApplication.retrofit
-                        val uploadService = retrofit.create(eventType.toServiceClass())
-                        val requestBody =
-                            RequestBody.create(MediaType.parse("multipart/form-data"), file)
-                        val part = MultipartBody.Part.createFormData("file", file.name, requestBody)
-                        val call = uploadService.uploadCsvFile(part)
+            Timber.tag("tuancoltech").v("Inside learningEventDir: ${learningEventDir.name}. Type: ${eventType.type}")
 
-                        Timber.i("call.request(): %s", call.request())
+
+            val files = learningEventDir.listFiles()
+            if (files != null) {
+                Timber.i("files.length: %s", files.size)
+                Arrays.sort(files)
+                for (i in files.indices) {
+                    val file = files[i]
+                    Timber.i("file.getAbsoluteFile(): %s %s", file.absoluteFile, "\nlearningEventDir: " + learningEventDir.name)
+                    Timber.i("file.getName(): %s", file.name)
+                    Timber.tag("tuancoltech").i("Uploading file: ${file.absoluteFile}")
+
+                    val baseApplication = applicationContext as BaseApplication
+                    val retrofit = baseApplication.retrofit
+                    val uploadService = retrofit.create(eventType.toServiceClass())
+                    val requestBody =
+                        RequestBody.create(MediaType.parse("multipart/form-data"), file)
+                    val part = MultipartBody.Part.createFormData("file", file.name, requestBody)
+                    val call = uploadService.uploadCsvFile(part)
+
+                    Timber.i("call.request(): %s", call.request())
+                    try {
+                        val response = call.execute()
+                        Timber.i("response: $response")
+                        Timber.i("response.isSuccessful(): %s", response.isSuccessful)
+                        if (response.isSuccessful) {
+                            val bodyString = response.body()?.string()
+                            Timber.i("bodyString: $bodyString")
+                        } else {
+                            val errorBodyString = response.errorBody()?.string()
+                            Timber.e("errorBodyString: $errorBodyString")
+                            // TODO: Handle error
+                        }
+                    } catch (e: IOException) {
+                        Timber.e(e)
+                    }
+
+                    // Delete log files which are older than 7 days in the past
```

Comment:
@tuancoltech Deleting previously generated CSV exports is out of scope for this PR's issue #291 ("Skip version code folder during CSV export"), so would you mind creating a new GitHub issue for handling deletion of CSVs from the file system? We need to think through the various edge-cases before implementing it.

Reply:
@jo-elimu Sure. 
@coderabbitai Help create a ticket for handling deletions of `.csv` log file which is older than 7 days from file system.